File: CS/WebApplication1/CustomControls/CustomAppointmentFormController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Internal;
using DevExpress.XtraScheduler;

namespace WebApplication1.CustomControls {
    public class CustomAppointmentFormController : AppointmentFormController {
        public CustomAppointmentFormController(ASPxScheduler control, Appointment apt) : base(control, apt) {
        
        }

        public string CustomInfoField {
            get { return (string)EditedAppointmentCopy.CustomFields["AppointmentInfo"]; }
            set { EditedAppointmentCopy.CustomFields["AppointmentInfo"] = value; }
        }
        string SourceCustomInfoField {
            get { return (string)SourceAppointment.CustomFields["AppointmentInfo"]; }
            set { SourceAppointment.CustomFields["AppointmentInfo"] = value; }
        }

        public byte[] CustomAttachment {
            get { return (byte[])EditedAppointmentCopy.CustomFields["AppointmentAttachment"]; }
            set { EditedAppointmentCopy.CustomFields["AppointmentAttachment"] = value; }
        }

        byte[] SourceCustomAttachment {
            get { return (byte[])SourceAppointment.CustomFields["AppointmentAttachment"]; }
            set { SourceAppointment.CustomFields["AppointmentAttachment"] = value; }
        }

        public string CustomAttachmentFileName {
            get { return (string)EditedAppointmentCopy.CustomFields["AppointmentAttachmentFileName"]; }
            set { EditedAppointmentCopy.CustomFields["AppointmentAttachmentFileName"] = value; }
        }

        string SourceCustomAttachmentFileName {
            get { return (string)SourceAppointment.CustomFields["AppointmentAttachmentFileName"]; }
            set { SourceAppointment.CustomFields["AppointmentAttachmentFileName"] = value; }
        }
    }
}

File: VB/WebApplication1/CustomControls/CustomAppointmentFormController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Internal
Imports DevExpress.XtraScheduler

Namespace WebApplication1.CustomControls
	Public Class CustomAppointmentFormController
		Inherits AppointmentFormController
		Public Sub New(ByVal control As ASPxScheduler, ByVal apt As Appointment)
			MyBase.New(control, apt)

		End Sub

		Public Property CustomInfoField() As String
			Get
				Return CStr(EditedAppointmentCopy.CustomFields("AppointmentInfo"))
			End Get
			Set(ByVal value As String)
				EditedAppointmentCopy.CustomFields("AppointmentInfo") = value
			End Set
		End Property
		Private Property SourceCustomInfoField() As String
			Get
				Return CStr(SourceAppointment.CustomFields("AppointmentInfo"))
			End Get
			Set(ByVal value As String)
				SourceAppointment.CustomFields("AppointmentInfo") = value
			End Set
		End Property

		Public Property CustomAttachment() As Byte()
			Get
				Return CType(EditedAppointmentCopy.CustomFields("AppointmentAttachment"), Byte())
			End Get
			Set(ByVal value As Byte())
				EditedAppointmentCopy.CustomFields("AppointmentAttachment") = value
			End Set
		End Property

		Private Property SourceCustomAttachment() As Byte()
			Get
				Return CType(SourceAppointment.CustomFields("AppointmentAttachment"), Byte())
			End Get
			Set(ByVal value As Byte())
				SourceAppointment.CustomFields("AppointmentAttachment") = value
			End Set
		End Property

		Public Property CustomAttachmentFileName() As String
			Get
				Return CStr(EditedAppointmentCopy.CustomFields("AppointmentAttachmentFileName"))
			End Get
			Set(ByVal value As String)
				EditedAppointmentCopy.CustomFields("AppointmentAttachmentFileName") = value
			End Set
		End Property

		Private Property SourceCustomAttachmentFileName() As String
			Get
				Return CStr(SourceAppointment.CustomFields("AppointmentAttachmentFileName"))
			End Get
			Set(ByVal value As String)
				SourceAppointment.CustomFields("AppointmentAttachmentFileName") = value
			End Set
		End Property
	End Class
End Namespace